crates/sui-core/src/authority.rs

```diff
@@ -1912,6 +1913,11 @@ impl AuthorityState {
                 .db_checkpoint_config
                 .perform_db_checkpoints_at_epoch_end
             {
+                DBCheckpointWaiter::wait_until_num_pending_is_less_than_threshold(
```

Comment:
If we can tolerate arbitrary wait times on nodes that are uploading snapshots to S3, why not perform a full compaction for the main db? It should reduce the overall amount of work. Currently, for every checkpoint, we are compacting the full database from scratch and redoing the same work